docs/kubernetes/overlays.md

```diff
@@ -0,0 +1,125 @@
+# Overview of per kubernetes minor version overlays
+
+The Persistent Disk CSI Driver manifest [bundle](../../deploy/kubernetes/base) is a set of many kubernetes resources (driver pod which includes the containers csi-provisioner, csi-resizer, csi-snapshotter, gce-pd-driver, csi-driver-registrar; csi driver object, rbacs, pod security policies etc). Not all driver capabilties can be used with all kubernetes versions. For example [volume snapshots](https://kubernetes.io/docs/concepts/storage/volume-snapshots/) are supported on 1.17+ kubernetes versions. Thus structuring the overlays on a per kubernetes minor version is beneficial to quickly identify driver capabilities supported for a given minor version of kubernetes (example [stable-1-16](../../deploy/kubernetes/overlays/stable-1-16) driver manifests does not contain the snapshotter sidecar), and facilitates easy maintenance/updates of the CSI driver. Most up-to-date overlays is maintained in the master branch [overlays](https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/tree/master/deploy/kubernetes/overlays) directory. Every stable-{k8s-version} overlay has its staging counterpart prow-gke-staging-release-rc-{k8s-version}.
+
+Example:
+`stable-1-19` [overlays](../../deploy/kubernetes/overlays/stable-1-19) bundle can be used to deploy all the components of the driver on kubernetes 1.19.
+`prow-gke-staging-release-rc-1-19` [overlays](../../deploy/kubernetes/overlays/prow-gke-staging-release-rc-1-19) bundle can be used to deploy all the components of the driver on kubernetes 1.19.
+
+## General guidelines for creation/modification of new/existing overlays
+
+When it comes to creation of overlays there are no strict rules on how overlays must be created/modified but following captures some of the common scenarios of adding new capabilities to the CSI driver bundle and how it would affect the overlays creation/modification.
+
+1. Sidecar image version update for existing overlays
+   * Say we have three existing stable overlays `stable-1-17`, `stable-1-18`, `stable-master`. Sidecar S1 is available 1.17+, S2 is avialable 1.18+. If we need to upgrade the image versions for these sidecars, we would do the following:
+   * Update the `deploy/kubernetes/images/prow-gke-staging-release-rc-1-17` for sidecar S1, `deploy/kubernetes/images/prow-gke-staging-release-rc-1-18` and `deploy/kubernetes/images/prow-gke-staging-release-rc-master` for S1 and S2.
+   * Validate the changes
```

Comment:
"... by verifying the [upstream testgrids](...) as well as verifying with repository maintainers that downstream testgrids are green."

(or whatever you think is appropriate, let's just be concrete about what validation means)